packages/next/src/server/web/adapter.ts

```diff
@@ -186,7 +186,8 @@ export async function adapter(
   let cookiesFromResponse
 
   // we only care to make async storage available for middleware
-  if (params.page === '/middleware') {
+  const isMiddleware = !!params.request.headers['x-middleware-invoke']
```

Comment:
We can't rely on this header when deployed to misc environments like serverless. Can we normalize the `params.page` value instead?

Reply:
```ts
const isMiddleware = params.page.includes('middleware')
```

Perhaps this could be good to go?

Reply:
I think we'll probably need to be explicit about the check since we don't want to accidentally match a page that contains `middleware` in the name